smart_open/gcs.py

```diff
@@ -91,6 +91,25 @@ def __init__(self, message, status_code, text):
         self.status_code = status_code
         self.text = text
 
+    @classmethod
+    def from_response(cls, response, part_num, content_length, total_size, headers):
+        msg = (
+            "upload failed ("
```

Comment:
This line splitting is a bit excessive.

You can also simplify the formatting code by setting the required local variables and then using `locals()`.